### ISSUE TITLE
[TASK] Investigate ways to implement modern authentication for email in contributionsBB

### ISSUE THREAD

**Issue by minump** [rokwire/rokwire-building-blocks-api#966]
### Description
Currently contributionsBB email notification uses smtp email server with username and password for authentication. This will be blocked after Aug 1st and OAuth2 protocols will be necessary. Check this link :  https://answers.uillinois.edu/illinois/page.php?id=47591

### Acceptance Criteria
Implementing modern authentication for email notifications

**Comment by minump** [comment]
Gmail supports OAuth2. Will need to switch from Office365 to gmail.
Useful resources for implementation are given below: 
Run-through for using python library [oauth2](https://github.com/google/gmail-oauth2-tools/wiki/OAuth2DotPyRunThrough)
[Blog](https://blog.macuyiko.com/post/2016/how-to-send-html-mails-with-oauth2-and-gmail-in-python.html) on how to send html mails with oauth2 and gmail.
[Github](https://github.com/google/gmail-oauth2-tools) repo for gmail oauth tools
OAuth2 [documentation](https://developers.google.com/gmail/imap/xoauth2-protocol)

**Comment by minump** [comment]
The current SMTP service can be used. However, will need to switch to [Cloud email services](https://answers.uillinois.edu/illinois/page.php?id=85362) later. 
Since the current SMTP services will not be affected / blocked by August 1st, moving this to low priority task.

**Comment by minump** [comment]
This is kept on hold until further developments / discussions.
